src/logger/loggerApi.c

```diff
@@ -1757,26 +1732,77 @@ int api_set_wifi_ap_cfg(struct Serial *s, const jsmntok_t *json)
         /* Inform the Wifi device that settings may have changed */
         wifi_update_ap_config(cfg);
 
-        return API_SUCCESS;
+        return true;
 }
 
-int api_get_wifi_ap_cfg(struct Serial *serial, const jsmntok_t *json)
+int api_set_wifi_cfg(struct Serial *serial, const jsmntok_t *json)
 {
         LoggerConfig *lc = getWorkingLoggerConfig();
-        const struct wifi_ap_cfg *cfg = &lc->ConnectivityConfigs.wifi.ap;
+        struct wifi_cfg *cfg = &lc->ConnectivityConfigs.wifi;
+        struct wifi_client_cfg *client_cfg = &cfg->client;
+        struct wifi_ap_cfg *ap_cfg = &cfg->ap;
 
-        json_objStart(serial);
-        json_objStartString(serial, "wifi_ap_cfg");
+        const jsmntok_t* client_json_root = jsmn_find_node(json, "client");
+        const jsmntok_t* ap_json_root = jsmn_find_node(json, "ap");;
+
+        if (ap_json_root)
+                if (!set_wifi_ap_cfg(ap_json_root, ap_cfg))
+                        return API_ERROR_PARAMETER;
+
+        if (client_json_root)
+                set_wifi_client_cfg(client_json_root, client_cfg);
+
+        setBoolValueIfExists(json, "active", &cfg->active);
+
+        return API_SUCCESS;
+}
+
+static void get_wifi_client_cfg(struct Serial *serial,
+                                const struct wifi_client_cfg* cfg,
+                                const bool more)
+
+{
+        json_objStartString(serial, "client");
+
+        json_bool(serial, "active", cfg->active, true);
+        json_string(serial, "ssid", cfg->ssid, true);
+        json_string(serial, "password", cfg->passwd, false);
+
+        json_objEnd(serial, more);
+}
+
+static void get_wifi_ap_cfg(struct Serial *serial,
+                            const struct wifi_ap_cfg* cfg,
+                            const bool more)
+{
+        json_objStartString(serial, "ap");
 
         json_bool(serial, "active", cfg->active, 1);
-        json_string(serial, "ssid", cfg->ssid,1);
-        json_string(serial, "passwd", cfg->password,1);
-        json_int(serial, "channel", cfg->channel,1);
+        json_string(serial, "ssid", cfg->ssid, 1);
+        json_string(serial, "password", cfg->password, 1);
+        json_int(serial, "channel", cfg->channel, 1);
         json_string(serial, "encryption",
                     wifi_api_get_encryption_str_val(cfg->encryption), 0);
 
-        json_objEnd(serial, 0);
-        json_objEnd(serial, 0);
+        json_objEnd(serial, more);
+}
+
+int api_get_wifi_cfg(struct Serial *serial, const jsmntok_t *json)
+{
+        const LoggerConfig *lc = getWorkingLoggerConfig();
+        const struct wifi_cfg *cfg = &lc->ConnectivityConfigs.wifi;
+        const struct wifi_client_cfg *client_cfg = &cfg->client;
+        const struct wifi_ap_cfg *ap_cfg = &cfg->ap;
+
+        json_objStart(serial);
+        json_objStartString(serial, "getWifiCfg");
```

Comment:
to be consistent with other APIs this json object's root name should be: wifiCfg